rotkehlchen/accounting/pot.py

```diff
@@ -349,49 +357,53 @@ def get_prices_for_swap(
             return None
 
         try:
-            out_price = self.get_rate_in_profit_currency(
+            oracle_out_price = self.get_rate_in_profit_currency(
                 asset=asset_out,
                 timestamp=timestamp,
             )
         except (PriceQueryUnsupportedAsset, NoPriceForGivenTimestamp, RemoteError):
-            out_price = None
+            oracle_out_price = None
 
-        fee_price = None
-        if fee is not None and fee_asset is not None and fee != ZERO:
-            # also checking fee_asset != None due to https://github.com/rotki/rotki/issues/4172
+        oracle_fee_price = None
+        if fee_info is not None:
             try:
-                fee_price = self.get_rate_in_profit_currency(
-                    asset=fee_asset,
+                oracle_fee_price = self.get_rate_in_profit_currency(
+                    asset=fee_info[1],
                     timestamp=timestamp,
                 )
-            except (PriceQueryUnsupportedAsset, NoPriceForGivenTimestamp, RemoteError):
-                fee_price = None
+            except (PriceQueryUnsupportedAsset, RemoteError):
+                oracle_fee_price = None
+            except NoPriceForGivenTimestamp:
+                raise
+
         try:
-            in_price = self.get_rate_in_profit_currency(
+            oracle_in_price = self.get_rate_in_profit_currency(
                 asset=asset_in,
                 timestamp=timestamp,
             )
         except (PriceQueryUnsupportedAsset, RemoteError):
-            in_price = None
+            oracle_in_price = None
         except NoPriceForGivenTimestamp as e:
-            in_price = None
-            if e.rate_limited is True and out_price is None:
+            oracle_in_price = None
+            if e.rate_limited is True and oracle_out_price is None:
                 raise  # in_price = out_price = None -> notify user
 
-        if out_price is None and in_price is None:
+        if oracle_out_price is None and oracle_in_price is None:
             return None
 
-        if out_price is not None:
-            paid = amount_out * out_price
-            if fee_price is not None:
-                paid += fee_price * fee  # type: ignore # fee should exist here
-            calculated_in = Price(paid / amount_in)
+        if oracle_out_price is not None:
+            total_paid = amount_out * oracle_out_price  # Use calculated_out_price
         else:
-            calculated_in = in_price  # type: ignore # in_price should exist here
+            total_paid = amount_in * oracle_in_price  # type: ignore  # in_price should exist here  # noqa: E501
```

Comment:
Here and in other parts in this code you should rename in_price to oracle_in_price and out price
```suggestion
            total_paid = amount_in * oracle_in_price  # type: ignore  # oracle_in_price should exist here  # noqa: E501
```

But why add the `oracle_` prefix everywhere? It just makes it all longer.